### ISSUE TITLE
Replication applier sometimes does not start on system reboots

### ISSUE THREAD

**Issue by F21** [arangodb/arangodb#1557]
I am finding that sometimes, if I reboot an arangodb replication slave, the replication applier does not start once the machine is rebooted even when I set `autostart` to `true`.

I am using the following to setup replication:

```
curl -XPUT  -d '{"endpoint":"tcp://arangodb.master:8529", "database":"graphs", "maxConnectRetries":1000, "autoStart":true, "adaptivePolling":true, "includeSystem":true}' http://127.0.0.1:8529/_db/graphs/_api/replication/applier-config

curl -XPUT  -d '{"endpoint":"tcp://arangodb.master:8529", "database":"graphs"}' http://127.0.0.1:8529/_db/graphs/_api/replication/sync

curl -XPUT http://127.0.0.1:8529/_db/graphs/_api/replication/applier-start?from={last_log_tick_here}
```

It doesn't happen all the time, but it is noticeable.

ArangoDB is 2.6.9 running on Ubuntu 15.04 64-bit. 

Here are the relevant log entries:

```
2015-11-02T21:34:46Z [669] INFO ArangoDB 2.6.9 64bit -- ICU 54.1, V8 4.1.0.27, OpenSSL 1.0.1f 6 Jan 2014
2015-11-02T21:34:46Z [669] INFO using default language 'en'
2015-11-02T21:34:46Z [669] INFO loaded database '_system' from '/var/lib/arangodb/databases/database-1499160504100'
2015-11-02T21:34:46Z [669] INFO loaded database 'graphs' from '/var/lib/arangodb/databases/database-1499192320538'
2015-11-02T21:34:46Z [669] INFO JavaScript using startup '/usr/share/arangodb/js', application '/var/lib/arangodb-apps'
2015-11-02T21:35:01Z [1373] INFO ArangoDB 2.6.9 64bit -- ICU 54.1, V8 4.1.0.27, OpenSSL 1.0.1f 6 Jan 2014
2015-11-02T21:35:01Z [1373] INFO using default language 'en'
2015-11-02T21:35:01Z [1373] INFO starting up in supervisor mode
2015-11-02T21:35:01Z [1375] INFO loaded database '_system' from '/var/lib/arangodb/databases/database-1499160504100'
2015-11-02T21:35:01Z [1375] INFO loaded database 'graphs' from '/var/lib/arangodb/databases/database-1499192320538'
2015-11-02T21:35:01Z [1375] INFO running WAL recovery (1 logfiles)
2015-11-02T21:35:01Z [1375] INFO replaying WAL logfile '/var/lib/arangodb/journals/logfile-1499169972762.db' (1 of 1)
2015-11-02T21:35:01Z [1375] INFO WAL recovery finished successfully
2015-11-02T21:35:01Z [1375] INFO started replication applier for database 'graphs'
2015-11-02T21:35:01Z [1375] INFO using endpoint 'tcp://127.0.0.1:8529' for non-encrypted requests
2015-11-02T21:35:01Z [1375] INFO using default API compatibility: 20600
2015-11-02T21:35:01Z [1375] INFO JavaScript using startup '/usr/share/arangodb/js', application '/var/lib/arangodb-apps'
2015-11-02T21:35:02Z [1375] INFO connected to master at tcp://10.8.0.1:8529, id 184573421207002, version 2.6, last log tick 1465332193621
2015-11-02T21:35:02Z [1375] ERROR replication applier error for database 'graphs': no start tick
2015-11-02T21:35:02Z [1375] INFO stopped replication applier for database 'graphs'
2015-11-02T21:35:10Z [1375] INFO In database 'graphs': Database is up-to-date (20609/prod/standalone/existing)
2015-11-02T21:35:10Z [1375] INFO In database '_system': Database is up-to-date (20609/prod/standalone/existing)
2015-11-02T21:35:11Z [1375] INFO Please note that a new minor version '2.7.0' is available
2015-11-02T21:35:12Z [1375] INFO permanently changing the gid to 999
2015-11-02T21:35:12Z [1375] INFO permanently changing the uid to 999
2015-11-02T21:35:12Z [1375] INFO Authentication is turned off
2015-11-02T21:35:12Z [1375] INFO ArangoDB (version 2.6.9 [linux]) is ready for business. Have fun!
```

**Comment by dothebart** [comment]
We already found that. There is a bug about saving the replication state when creating a new replication and restarting the server without ever transfering a single document.
Please work around it by simply adding a document in some collection on the master.
You will be able to find in the databases/database-xxx/REPLICATION-APPLIER-STATE whether the position is adjusted properly.

Once the replication has started transfering actual new items, it should work finde though.

**Comment by jsteemann** [comment]
The problem occurred under the following circumstances: 
* a slave server was created and the continuous replication was set up for it using a start tick
* the slave received no data from the master via the continuous replication (i.e. no new data from master after `require("org/arangodb/replication").applier.start(lastLogTick);` was run.
* slave server was shut down and re-started

In this case, the slave server restarted but found that it had "no start tick". The reason for this is that the "start tick" state was not properly saved when the slave never received any new data from the master via the continuous replication.
This has been fixed in 2.7 and 2.8 and the fix will be included in 2.7.1 and 2.8.

A workaround for 2.6 is to change data on the master so the slave can pick it up and update its tick state after the continuous replication applier was started. This only needs to be done once after the applier was initially started, and not once per slave re-start. Note however that there can be multiple replication appliers in a single ArangoDB instance.

**Comment by jsteemann** [comment]
Should be fixed from 2.7 onwards, at least when the replication applier did not fail before with a non-recoverable error such as a unique constraint violation.
